src/pep8ext_naming.py

```diff
@@ -275,6 +276,29 @@ def superclass_names(cls, name, parents: Iterable, _names=None):
                 names.update(cls.superclass_names(base.id, parents, names))
         return names
 
+    def visit_module(self, node, parents: Iterable, ignore=None):
+        for body in node.body:
+            try:
+                if not len(body.targets) == 1:
+                    continue
+                name = body.targets[0].id
+                func_name = body.value.func.id
+            except AttributeError:
+                continue
+
+            if _ignored(name, ignore) or func_name != "TypeVar":
+                return
+
+            if not name[:1].isupper():
+                yield self.err(body, 'N808', name=name)
+            elif '_' in name:
+                underscore_split = name.split('_')
+                suffix = underscore_split[-1]
+                if len(underscore_split) > 2 or (suffix != 'co' and suffix != 'contra'):
```

Comment:
I think this might be a little clearer if it focuses on a single `suffix` value:

```suggestion
                parts = name.split('_')
                suffix = parts[-1] if len(parts) > 2 else ''
                if suffix and suffix != 'co' and suffix != 'contra':
```